src/spin_mutex.cc

```diff
@@ -3,19 +3,40 @@
 #include <cstddef>
 #include <thread>
 
+#define _ASM_NOOP asm volatile("nop")
```

Comment:
If you can, put a ifdef x86 around, with warning in else, to make it compile on other arch but with reduced perf.

Reply:
We have rdtsc elsewhere and if this is not x86, it won't compile anyways. I prefer to fix that once we really plan to work on other ISA.

Reply:
Either way is fine, I would probably say less is better if checking x86 is not too hard.
If there's no good way to do that across compilers, let's forget about it.